src/core/swap.c

```diff
@@ -295,26 +295,22 @@ static int swap_verify(Swap *s) {
 }
 
 static int swap_load_devnode(Swap *s) {
-        _cleanup_(sd_device_unrefp) sd_device *d = NULL;
+        _cleanup_free_ char *p = NULL;
         struct stat st;
-        const char *p;
         int r;
 
         assert(s);
 
         if (stat(s->what, &st) < 0 || !S_ISBLK(st.st_mode))
                 return 0;
 
-        r = sd_device_new_from_stat_rdev(&d, &st);
+        r = devpath_from_devnum(S_IFBLK, st.st_rdev, &p);
```

Comment:
maybe we should have devpath_from_stat_rdev()? Much like we have sd_device_new_from_stat_rdev() as wrapper for device_new_from_mode_and_devnum()